src/main/java/com/dao/impl/orm/OrmAbstractDAO.java

```diff
@@ -0,0 +1,81 @@
+package com.dao.impl.orm;
+
+import com.dao.AbstractDAO;
+import com.dao.entity.IEntity;
+import com.dao.exception.DAOException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public abstract class OrmAbstractDAO<E extends IEntity> extends AbstractDAO<E> {
+    protected SessionFactory sessionFactory;
+
+    @Autowired
+    public void SetSessionFactory(SessionFactory sessionFactory) {
```

Comment:
Use _EntityManager_ instead of _SessionFactory_, this is more common way for persistence